src/cmd/services/m3comparator/main/querier_test.go

```diff
@@ -63,47 +63,89 @@ const (
 	defaultResolution = time.Second * 30
 )
 
-func TestFetchCompressedReturnsPreloadedData(t *testing.T) {
-	ctrl := xtest.NewController(t)
-	defer ctrl.Finish()
-
+func TestFetchCompressed(t *testing.T) {
 	const (
 		metricsName           = "preloaded"
 		predefinedSeriesCount = 10
 	)
 
-	query := matcherQuery(t, metricNameTag, metricsName)
-
-	metricsTag := ident.NewTags(ident.Tag{
-		Name:  ident.BytesID(tagOptions.MetricName()),
-		Value: ident.BytesID(metricsName),
-	})
-
-	iters := make([]encoding.SeriesIterator, 0, predefinedSeriesCount)
-	for i := 0; i < predefinedSeriesCount; i++ {
-		iters = append(iters, encoding.NewSeriesIterator(
-			encoding.SeriesIteratorOptions{
-				Namespace:      ident.StringID("ns"),
-				Tags:           ident.NewTagsIterator(metricsTag),
-				StartInclusive: xtime.ToUnixNano(query.Start),
-				EndExclusive:   xtime.ToUnixNano(query.End),
-			}, nil))
+	tests := []struct {
+		name          string
+		queryTagName  string
+		queryTagValue string
+		expectedCount int
+	}{
+		{
+			name:          "querying by metric name returns preloaded data",
+			queryTagName:  metricNameTag,
+			queryTagValue: metricsName,
+			expectedCount: predefinedSeriesCount,
+		},
+		{
+			name:          "querying without metric name just by other tag returns preloaded data",
+			queryTagName:  "tag1",
+			queryTagValue: "test2",
+			expectedCount: 4,
+		},
 	}
 
-	seriesIterators := encoding.NewMockSeriesIterators(ctrl)
-	seriesIterators.EXPECT().Len().Return(predefinedSeriesCount).MinTimes(1)
-	seriesIterators.EXPECT().Iters().Return(iters).Times(1)
-	seriesIterators.EXPECT().Close()
+	for _, tt := range tests {
+		t.Run(tt.name, func(t *testing.T) {
+			ctrl := xtest.NewController(t)
+			defer ctrl.Finish()
+
+			query := matcherQuery(t, tt.queryTagName, tt.queryTagValue)
 
-	seriesLoader := &testSeriesLoadHandler{seriesIterators}
+			metricsTag := ident.NewTags(ident.Tag{
+				Name:  ident.BytesID(tagOptions.MetricName()),
+				Value: ident.BytesID(metricsName),
+			},
+				ident.Tag{
+					Name:  ident.BytesID("tag1"),
+					Value: ident.BytesID("test"),
+				},
+			)
+			metricsTag2 := ident.NewTags(ident.Tag{
+				Name:  ident.BytesID(tagOptions.MetricName()),
+				Value: ident.BytesID(metricsName),
+			},
+				ident.Tag{
+					Name:  ident.BytesID("tag1"),
+					Value: ident.BytesID("test2"),
+				},
+			)
+
+			iters := make([]encoding.SeriesIterator, 0, predefinedSeriesCount)
+			for i := 0; i < predefinedSeriesCount; i++ {
+				m := metricsTag
+				if i > 5 {
+					m = metricsTag2
+				}
```

Comment:
Somehow it feels that using test tables adds a lot of complexity overhead. Might be worth trying to do it with two separate test methods, and extracting all the shared boilerplate (such as `seriesIterators` mock setup) into a helper method.